avocado.test: Escape filename when executing SimpleTest

ldoktor opened the pull request:
Fixes the https://github.com/avocado-framework/avocado/issues/415

Current behavior of process.run(shell=False) is it expects multiple
arguments and split the provided cmd. This doesn't fit the SimpleTest
as it founds a binary and then tries to execute it without any
arguments as it might be separated by spaces. This patch escapes the
executed command using "pipes.quote" to avoid it's splitting.

Signed-off-by: Lukáš Doktor <ldoktor@redhat.com>

ldoktor commented:
Well now when I think about it, this could lead to hidden errors. For example executing "true" test if "true" is executable inside this directory would result in system `true` command instead. I'll replace this version with the one which escapes the command...

ruda commented on avocado/test.py:
```diff
@@ -11,7 +11,6 @@
 # This code was inspired in the autotest project,
 # client/shared/test.py
 # Authors: Martin J Bligh <mbligh@google.com>, Andy Whitcroft <apw@shadowen.org>
-
```
sorry for being pedantic: I would keep this space here, usually (in our code) there's a space between the comments and the module's docstring that follows.

ldoktor replied:
Sorry, it was reformated automatically by Eclipse... I'll change it in the next version.

ruda commented on avocado/test.py:
```diff
@@ -548,7 +548,8 @@ def action(self):
             test_params = {str(key): str(val)
                            for key, val in self.params.iteritems()}
-            result = process.run(self.path, verbose=True, env=test_params)
+            result = process.run(pipes.quote(self.path), verbose=True,
```
Thanks for pointing this bug, it's inconvenient to scape like this `process.run('/home/rmoura/true\ or\ false')`, let's do it, but I would like to see a test case for using space in the name of the command, so we can merge it.

lmr replied:
+1 for having a test for the bug mentioned.

ldoktor replied:
OK, I'll add the selftest on Monday. (but is reported in GH issues)

ldoktor commented:
Oups, I'm sorry I accidentally pushed the changes here. V0 commit is https://github.com/ldoktor/avocado/commit/d057eb9b1935d8728eaf79b8e0357c79b720d448

Changes:

*  removed the eclipse autoformated change
*  added note about the reason to escape the path

lmr commented:
Looks good, thanks @ldoktor !
